db/hints/manager.cc

```diff
@@ -185,15 +185,22 @@ future<> manager::start(shared_ptr<gms::gossiper> gossiper_ptr) {
 
     co_await lister::scan_dir(_hints_dir, lister::dir_entry_types::of<directory_entry_type::directory>(),
             [this] (fs::path datadir, directory_entry de) {
-        endpoint_id ep = endpoint_id{de.name};
+        const auto maybe_ep = std::invoke([&] () -> std::optional<endpoint_id> {
+            try {
+                return endpoint_id{de.name};
+            } catch (...) {
+                manager_logger.warn("Encountered a hint directory of invalid name: {}. It will be ignored.", de.name);
```

Comment:
> `"Encountered a hint directory of invalid name: {}. It will be ignored."`

I'd concentrate more on the consequences to the user, i.e. what it means that the directory will be ignored. In addition to the current message, I'd stress that hints from this directory will not be replayed.